lua/ttt2/libraries/vguihandler.lua

```diff
@@ -0,0 +1,167 @@
+---
+-- A vgui handler
+-- @author Mineotopia
+
+AddCSLuaFile()
+
+-- the rest of the vguihandler library is client only
+if SERVER then return end
+
+local table = table
+
+vguihandler = vguihandler or {
+	frames = {},
+	callback = {
+		frame = {},
+		hidden = {}
+	}
+}
+
+-- Call this function to create a new frame. Using this instead of setting the
+-- frame up manually has the benefit that all basic settings are already set.
+-- Additionally the frame is added to a managed list that is used to handle
+-- hiding, closing and tracking menues.
+-- @param number w The width of the panel
+-- @param number h The height of the panel
+-- @param string title The title of the panel
+-- @return @{Panel} The created/cleared DFrameTTT2 object
+function vguihandler.GenerateFrame(w, h, title)
+	local frame = vgui.Create("DFrameTTT2")
+
+	frame:SetSize(w, h)
+	frame:Center()
+	frame:SetTitle(title)
+
+	local OriginalClose = frame.Close
+
+	frame.Close = function(slf)
+		if slf:GetDeleteOnClose() then
+			table.RemoveByValue(vguihandler.frames, frame)
+		end
+
+		OriginalClose(slf)
+	end
+
+	vguihandler.frames[#vguihandler.frames + 1] = frame
+
+	return frame
+end
+
+---
+-- Hides all registered and unhidden frames.
+-- @return table Returns a table of the frames that are now hidden
+-- @realm client
+function vguihandler.HideUnhiddenFrames()
+	local frames = vguihandler.frames
+	local hiddenFrames = {}
+
+	for i = 1, #frames do
+		local frame = frames[i]
+
+		if frame:IsFrameHidden() then continue end
+
+		frame:HideFrame()
+
+		hiddenFrames[#hiddenFrames + 1] = frame
+	end
+
+	return hiddenFrames
+end
+
+---
+-- Unhides frames that are listed in a table.
+-- @param table A table of frames
+-- @realm client
+function vguihandler.UnhideFrames(frames)
+	for i = 1, #frames do
+		local frame = frames[i]
+
+		if not IsValid(frame) or not frame:IsFrameHidden() then continue end
+
+		frame:ShowFrame()
+	end
+end
+
+---
+-- Unhides all frames that are currently registered and hidden.
+-- @realm client
+function vguihandler.UnhideAllFrames()
+	local frames = vguihandler.frames
+
+	for i = 1, #frames do
+		local frame = frames[i]
+
+		if not frame:IsFrameHidden() then continue end
+
+		frame:ShowFrame()
+	end
+end
+
+---
+-- Should be called when a specific vskin variable is changed
+-- so that the complete vgui element is redone
+-- @param string name The name of the changed setting
+-- @internal
+-- @realm client
+function vguihandler.InvalidateVSkin()
+	local frames = vguihandler.frames
+
+	for i = 1, #frames do
+		frames[i]:InvalidateLayout()
+	end
+end
+
+---
+-- Rebuilds the whole menu without a specific changed setting.
+-- @realm client
+function vguihandler.Rebuild()
+	local frames = vguihandler.frames
+
+	for i = 1, #frames do
+		local frame = frames[i]
+
+		if isfunction(frame.OnRebuild) then
```

Comment:
`isfunction` check isn't needed here (you don't using it in many other places, regarding the function name, it's not expected/intended that this is not a function)

Reply:
Hm, I don't think removing it improves the code in any way.